duckduckgo/iOS
# Don't show key icon for empty passwords

Task/Issue URL: https://app.asana.com/0/1202926619870900/1207697076491402/f
Tech Design URL: https://app.asana.com/0/481882893211075/1207750849382981

**Description**:

There has been an issue around for some time where the autofill key icon (indicating that there is autofill data for that field) was showing in the password field despite there being no password stored for a domain. This was because we were using the existence of any stored data for that domain as the source of truth for showing the icon in both username and password fields.

**Steps to test this PR**:
1. Go to Setting -> Passwords
2. Make sure there are no other login for the domain https://fill.dev/
3. Add a login for the domain https://fill.dev/, filling in the username field but not the password field.
4. Visit https://fill.dev/
5. **Observe there is a key icon in the username field but not the password**
6. Repeat step 1
7. Find the login added in step 3 and add a password.
8. Refresh the https://fill.dev/ page.
9. **Observe there are key icons in both fields**

**Definition of Done (Internal Only)**:


**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
